### PR TITLE
displayname disambiguation fixes

### PR DESCRIPTION
**Pull request by t3chguy**
+ fixes disambiguation occurring if someone's is set to `undefined` because javascript type coercion.
+ strips zero width chars and whitespace from displayName when doing disambiguation logic
+ if someone's displayname is just spaces/zero width chars show just their MXID instead.

### Fixes https://github.com/vector-im/riot-web/issues/6896 and https://github.com/vector-im/riot-web/issues/5827

Signed-off-by: Michael Telatynski <7t3chguy@gmail.com>

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -664,10 +664,10 @@ module.exports.isNumber = function(value) {
 };
 
 /**
- * Strips zero width chars and padding spaces from the displayName
+ * Strips zero width chars, diaritics and whitespace from the displayName
```
Comment (lukebarnard1) [inline src/utils.js]: I think you mean "diacritics"

File: src/models/room-state.js [hunk 2]
```diff
@@ -494,22 +494,30 @@ function _updateDisplayNameCache(roomState, userId, displayName) {
...
-                // remove this user ID from this array
-                existingUserIds.splice(i, 1);
-                i--;
+        const strippedOldName = utils.stripDisplayName(oldName);
+        const existingUserIds = roomState._displayNameToUserIds[strippedOldName];
+        if (existingUserIds) {
+            for (let i = 0; i < existingUserIds.length; i++) {
+                if (existingUserIds[i] === userId) {
+                    // remove this user ID from this array
+                    existingUserIds.splice(i, 1);
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/models/room-state.js]: this less readable slice method is a lot more efficient though,
Maybe we should use a `Set` considering they're already used in the Crypto part of this SDK

[the remark this answers is not in the document]
Comment (lukebarnard1) [reply src/models/room-state.js]: Sorry, to be clear I meant `existingUserIDs.filter((id)  => id !== userId)` as think this is what the for loop is doing.

File: src/utils.js (hunk 1 above)
Comment (t3chguy) [reply src/utils.js]: Whoops, I rely on my IDE's spellcheck kicking in too heavily and it somehow didn't catch that :(

**Pull request merged**